src/main/steps/respondent/keep-details-private/KeepDetailsPrivatePostController.ts

```diff
@@ -0,0 +1,53 @@
+import autobind from 'autobind-decorator';
+import type { Response } from 'express';
+
+import { getSystemUser } from '../../../app/auth/user/oidc';
+import { CosApiClient } from '../../../app/case/CosApiClient';
+import { Respondent, YesOrNo } from '../../../app/case/definition';
+import { toApiFormat } from '../../../app/case/to-api-format';
+import { AppRequest } from '../../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../../app/controller/PostController';
+import { FormFields, FormFieldsFn } from '../../../app/form/Form';
+import { RESPONDENT_PRIVATE_DETAILS_CONFIRMED, RESPONDENT_PRIVATE_DETAILS_NOT_CONFIRMED } from '../../../steps/urls';
+
+import { setKeepYourDetailsPrivate } from './KeepYourDetailsPrivateMapper';
+
+@autobind
+export class KeepDetailsPrivatePostController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const caseworkerUser = await getSystemUser();
+    const caseReference = req.session.userCase.id;
+
+    const client = new CosApiClient(caseworkerUser.accessToken, 'https://return-url');
+
+    const caseDataFromCos = await client.retrieveByCaseId(caseReference, caseworkerUser);
+    Object.assign(req.session.userCase, caseDataFromCos);
```

Comment:
Do we need this call? if yes you can directly assign req.session.usercase to the response